xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp

```diff
@@ -473,8 +472,284 @@ bool CVaapiTexture::TestInteropHevc(VADisplay vaDpy, EGLDisplay eglDisplay)
     }
     vaDestroyImage(vaDpy, image.image_id);
   }
+
   vaDestroySurfaces(vaDpy, &surface, 1);
 
   return ret;
 }
 
+void CVaapi2Texture::Init(InteropInfo& interop)
+{
+  m_interop = interop;
+  m_hasPlaneModifiers = CEGLUtils::HasExtension(m_interop.eglDisplay, "EGL_EXT_image_dma_buf_import_modifiers");
+}
+
+bool CVaapi2Texture::Map(CVaapiRenderPicture* pic)
+{
+#if HAVE_VAEXPORTSURFACHEHANDLE
+  if (m_vaapiPic)
+    return true;
+
+  m_vaapiPic = pic;
+  m_vaapiPic->Acquire();
+
+  VAStatus status;
+
+  VADRMPRIMESurfaceDescriptor surface;
+
+  status = vaExportSurfaceHandle(pic->vadsp, pic->procPic.videoSurface,
+    VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
+    VA_EXPORT_SURFACE_READ_ONLY | VA_EXPORT_SURFACE_SEPARATE_LAYERS,
+    &surface);
+
+  if (status != VA_STATUS_SUCCESS)
+  {
+    CLog::LogFunction(LOGWARNING, "CVaapi2Texture::Map", "vaExportSurfaceHandle failed - Error: %s (%d)", vaErrorStr(status), status);
+    return false;
+  }
+
+  // Remember fds to close them later
+  if (surface.num_objects > m_drmFDs.size())
+    throw std::logic_error("Too many fds returned by vaExportSurfaceHandle");
+
+  for (int object = 0; object < surface.num_objects; object++)
+  {
+    m_drmFDs[object].attach(surface.objects[object].fd);
+  }
+
+  status = vaSyncSurface(pic->vadsp, pic->procPic.videoSurface);
+  if (status != VA_STATUS_SUCCESS)
+  {
+    CLog::LogFunction(LOGERROR, "CVaapi2Texture::Map", "vaSyncSurface - Error: %s (%d)", vaErrorStr(status), status);
+    return false;
+  }
+
+  m_textureSize.Set(pic->DVDPic.iWidth, pic->DVDPic.iHeight);
+
+  switch (surface.fourcc)
+  {
+    case VA_FOURCC_NV12:
+      m_bits = 8;
+      break;
+
+    case VA_FOURCC_P010:
+    case VA_FOURCC_P016:
+      m_bits = 16;
```

Comment:
this is not correct. keep it the same as Texture1

Reply:
OK. Then maybe you could document what the bits value is being used for ;-)

Reply:
Setting this to 10 here breaks hevc10 display for me, everything goes pink

Reply:
Ah sorry, didn't see the overwrite further down. Maybe you should remove this line: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp#L158 It's not being used and quite confusing